modules/exploits/x3/x3_adxsrv_cmd.rb

```diff
@@ -0,0 +1,347 @@
+#
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Sage X3 Administration Service Command Execution',
```

Comment:
It is possible that in the future there will be multiple command execution vulnerabilities in this service.

At a glance, it appears as though executing arbitrary commands is intended functionality.

In which case, a more suitable module name should reference the authentication bypass as the key component of the exploit, ie: `Sage X3 Administration Service Authentication Bypass Command Execution`. The filename should also roughly match the module name, ie: `x3_adxsrv_auth_byoass_cmd_exec.rb`.

Alternatively, optionally, if command execution is intended functionality, this module could by updated to support a username and password combination to gain code execution, and a separate option to override with auth bypass instead.

Reply:
Renamed file and updated name in [this commit](https://github.com/deadjakk/metasploit-framework/commit/fa98e9a3a407d740d5e6346ed2d1f2a683d5788c)